src/rust/en.nana/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+#![no_std]
+use aidoku::{
+	prelude::*, error::Result, std::String, std::Vec, std::net::Request, std::net::HttpMethod,
+	Filter, Manga, MangaPageResult, Page, Chapter
+};
+
+mod parser;
+
+const BASE_URL: &str = "https://nana.my.id";
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut result: Vec<Manga> = Vec::new();
+	let mut url = String::new();
+
+	parser::get_filtered_url(filters, page, &mut url);
+	
+	let html = Request::new(url.as_str(), HttpMethod::Get).html()?;
+	let html_next_page = Request::new(url.as_str(), HttpMethod::Get).html()?;
+
+	parser::parse_search(html, &mut result);
+
+	Ok(MangaPageResult {
+		manga: result,
+		has_more: html_next_page.select("a.paginate_button.current + a.paginate_button").array().len() > 0,
+	})
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn get_manga_details(manga_rid: i32) -> i32 {
+    let manga = aidoku::std::ValueRef::new(manga_rid).as_object().unwrap();
+	let url = format!("{}/reader/{}", BASE_URL, manga.get("id").as_string().unwrap().read());
+	let html = Request::new(url.as_str(), HttpMethod::Get).html().expect("Node");
```

Comment:
```suggestion
	let html = match Request::new(url.as_str(), HttpMethod::Get).html() {
	    Ok(req) => req,
	    Err(_) => return,
	};
```

Reply:
Like I said on discord though, you shouldn't perform this request if you already have both the cover and author. Check if you have those before doing this

Reply:
i couldn't use the return like this so i used a panic function is it okay to use it?

Reply:
> 

this one was for history page

Reply:
don't use panic. I meant to do `return -1` though.

> this one was for history page
the same function is used for both history page and opening the manga page from the list. there's no reason to send another request when opening from the list because you're not getting new info. if it's coming from the history page, the info (title and cover) won't exist so you can just check that to see.

Reply:
> the same function is used for both history page and opening the manga page from the list. there's no reason to send another request when opening from the list because you're not getting new info. if it's coming from the history page, the info (title and cover) won't exist so you can just check that to see.

if i didn't do it like this the thumbnails and cover wouldn't appear at all and what do you mean i need to make a check for something?

Reply:
if the cover and title exist, you don't need the request (you don't use it anyways -- cover and title already exist and you fetch them from the manga object, not the request object). If they don't exist, then you do the request to grab them.

Reply:
I did it like this, is this one better?

```rs
	let mut img_url = String::new();
	let mut title = String::new();

	if manga.get("title").as_string().is_err() && manga.get("cover").as_string().is_err() {
		let html = match Request::new(url.as_str(), HttpMethod::Get).html() {
			Ok(req) => req,
			Err(_) => return -1,
		};

		title.push_str(html.select("#archivePagesOverlay .spanh3reader").text().read().as_str().trim());

		let img = html.select("a#display img").attr("src").read();
		let edit_img_url = if img.starts_with('/') {
			format!("{}{}", BASE_URL, img)
		} else {
			img
		}.replace("/image/page", "/image/thumbnails");

		img_url.push_str(edit_img_url.as_str());
	}
```